From f98b6e3365fed5dee83c782bba8d13d2f8f3b792 Mon Sep 17 00:00:00 2001
From: kikairoya <kikairoya@gmail.com>
Date: Wed, 17 Sep 2025 07:02:26 +0900
Subject: [PATCH] add /lib/win32api

---
 clang/lib/Driver/ToolChains/Cygwin.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/clang/lib/Driver/ToolChains/Cygwin.cpp b/clang/lib/Driver/ToolChains/Cygwin.cpp
index b489ff39294f..f6666032fa8c 100644
--- a/clang/lib/Driver/ToolChains/Cygwin.cpp
+++ b/clang/lib/Driver/ToolChains/Cygwin.cpp
@@ -46,6 +46,7 @@ Cygwin::Cygwin(const Driver &D, const llvm::Triple &Triple, const ArgList &Args)
 
   addPathIfExists(D, SysRoot + "/lib", Paths);
   addPathIfExists(D, SysRoot + "/usr/lib", Paths);
+  addPathIfExists(D, SysRoot + "/lib/w32api", Paths);
   addPathIfExists(D, SysRoot + "/usr/lib/w32api", Paths);
 }
 
-- 
2.51.0

