From d56b6c25d685cfecbe506664bc3de041c4843105 Mon Sep 17 00:00:00 2001
From: kikairoya <kikairoya@gmail.com>
Date: Sun, 7 Sep 2025 22:12:34 +0900
Subject: [PATCH 13/28] update test

---
 .../CodeGenCXX/mingw-template-dllexport.cpp   | 20 ++++++++-----------
 1 file changed, 8 insertions(+), 12 deletions(-)

diff --git a/clang/test/CodeGenCXX/mingw-template-dllexport.cpp b/clang/test/CodeGenCXX/mingw-template-dllexport.cpp
index a0a411d09f659c5de0635eefc43496e0b7fdf6e9..ee90b6d81dc6371a7b8afdfe001c80a6dc2091c1 100644
--- a/clang/test/CodeGenCXX/mingw-template-dllexport.cpp
+++ b/clang/test/CodeGenCXX/mingw-template-dllexport.cpp
@@ -196,23 +196,19 @@ template struct __declspec(dllexport) outer<int>;
 // FUNC-O0:  declare{{.*}} dllimport{{.*}} @_ZN5outerIcEC1EOS0_
 // FUNC-O0:  declare{{.*}} dllimport{{.*}} @_ZN5outerIcEaSERKS0_
 // FUNC-O0:  declare{{.*}} dllimport{{.*}} @_ZN5outerIcEaSEOS0_
-// FUNC-O0:  define{{.*}} dso_local{{.*}} @_ZN5outerIcE5innerC1Ev
-// FUNC-O0:  define{{.*}} dso_local{{.*}} @_ZN5outerIcE5innerC1ERKS1_
-// FUNC-O0:  define{{.*}} dso_local{{.*}} @_ZN5outerIcE5innerC1EOS1_
-// FUNC-O0:  define{{.*}} dso_local{{.*}} @_ZN5outerIcE5inneraSERKS1_
-// FUNC-O0:  define{{.*}} dso_local{{.*}} @_ZN5outerIcE5inneraSEOS1_
+// FUNC-O0:  declare{{.*}} dso_local{{.*}} @_ZN5outerIcE5innerC1Ev
+// FUNC-O0:  declare{{.*}} dso_local{{.*}} @_ZN5outerIcE5innerC1ERKS1_
+// FUNC-O0:  declare{{.*}} dso_local{{.*}} @_ZN5outerIcE5innerC1EOS1_
+// FUNC-O0:  declare{{.*}} dso_local{{.*}} @_ZN5outerIcE5inneraSERKS1_
+// FUNC-O0:  declare{{.*}} dso_local{{.*}} @_ZN5outerIcE5inneraSEOS1_
 // FUNC-O0:  declare{{.*}} dllimport{{.*}} @_ZN5outerIcE1fEv
 // FUNC:     declare{{.*}} dllimport{{.*}} @_ZN5outerIcE1gEv
 // FUNC-O0:  define{{.*}} dso_local{{.*}} @_ZN5outerIcE1hEv
-// FUNC-O0:  define{{.*}} dso_local{{.*}} @_ZN5outerIcE5inner1fEv
-// FUNC-O0:  define{{.*}} dso_local{{.*}} @_ZN5outerIcE5inner1gEv
+// FUNC-O0:  declare{{.*}} dso_local{{.*}} @_ZN5outerIcE5inner1fEv
+// FUNC:     declare{{.*}} dso_local{{.*}} @_ZN5outerIcE5inner1gEv
 // FUNC-O0:  define{{.*}} dso_local{{.*}} @_ZN5outerIcE5inner1hEv
-// FUNC-O0:  define{{.*}} dso_local{{.*}} @_ZN5outerIcE5innerD1Ev
+// FUNC-O0:  declare{{.*}} dso_local{{.*}} @_ZN5outerIcE5innerD1Ev
 // FUNC-O0:  declare{{.*}} dllimport{{.*}} @_ZN5outerIcED1Ev
-// FUNC-O0:  define{{.*}} dso_local{{.*}} @_ZN5outerIcE5innerC2Ev
-// FUNC-O0:  define{{.*}} dso_local{{.*}} @_ZN5outerIcE5innerD2Ev
-// FUNC-O0:  define{{.*}} dso_local{{.*}} @_ZN5outerIcE5innerC2ERKS1_
-// FUNC-O0:  define{{.*}} dso_local{{.*}} @_ZN5outerIcE5innerC2EOS1_
 extern template struct __declspec(dllimport) outer<char>;
 
 int anchor() noexcept {
-- 
2.51.2

