From 557eeec2ff67f8ed0466749ac7d725d93b2b01a1 Mon Sep 17 00:00:00 2001
From: kikairoya <kikairoya@gmail.com>
Date: Fri, 27 Feb 2026 19:22:23 +0900
Subject: [PATCH 06/51] unfold

---
 ...rom_explicit_instantiation.nontemplate.cpp | 57 +++++++------------
 1 file changed, 19 insertions(+), 38 deletions(-)

diff --git a/clang/test/SemaCXX/attr-exclude_from_explicit_instantiation.nontemplate.cpp b/clang/test/SemaCXX/attr-exclude_from_explicit_instantiation.nontemplate.cpp
index 0196869117ed8ccf82bc204566affacc5f188dec..2a9c3fbef037f86229266278f8ddd2090d79d6fe 100644
--- a/clang/test/SemaCXX/attr-exclude_from_explicit_instantiation.nontemplate.cpp
+++ b/clang/test/SemaCXX/attr-exclude_from_explicit_instantiation.nontemplate.cpp
@@ -6,63 +6,44 @@
 #define EXCLUDE_ATTR __attribute__((exclude_from_explicit_instantiation))
 
 struct C {
-  EXCLUDE_ATTR void fn_excluded();
-  // expected-warning@-1{{'exclude_from_explicit_instantiation' attribute ignored on a non-template context}}
-  EXCLUDE_ATTR static int var_excluded;
-  // expected-warning@-1{{'exclude_from_explicit_instantiation' attribute ignored on a non-template context}}
-  struct EXCLUDE_ATTR nested_excluded {
-  // expected-warning@-1{{'exclude_from_explicit_instantiation' attribute ignored on a non-template context}}
-    EXCLUDE_ATTR void fn_excluded();
-    // expected-warning@-1{{'exclude_from_explicit_instantiation' attribute ignored on a non-template context}}
-    EXCLUDE_ATTR static int var_excluded;
-    // expected-warning@-1{{'exclude_from_explicit_instantiation' attribute ignored on a non-template context}}
+  EXCLUDE_ATTR void fn_excluded(); // expected-warning{{'exclude_from_explicit_instantiation' attribute ignored on a non-template context}}
+  EXCLUDE_ATTR static int var_excluded; // expected-warning{{'exclude_from_explicit_instantiation' attribute ignored on a non-template context}}
+  struct EXCLUDE_ATTR nested_excluded { // expected-warning{{'exclude_from_explicit_instantiation' attribute ignored on a non-template context}}
+    EXCLUDE_ATTR void fn_excluded(); // expected-warning{{'exclude_from_explicit_instantiation' attribute ignored on a non-template context}}
+    EXCLUDE_ATTR static int var_excluded; // expected-warning{{'exclude_from_explicit_instantiation' attribute ignored on a non-template context}}
   };
   struct nested {
-    EXCLUDE_ATTR void fn_excluded();
-    // expected-warning@-1{{'exclude_from_explicit_instantiation' attribute ignored on a non-template context}}
-    EXCLUDE_ATTR static int var_excluded;
-    // expected-warning@-1{{'exclude_from_explicit_instantiation' attribute ignored on a non-template context}}
+    EXCLUDE_ATTR void fn_excluded(); // expected-warning{{'exclude_from_explicit_instantiation' attribute ignored on a non-template context}}
+    EXCLUDE_ATTR static int var_excluded;  // expected-warning{{'exclude_from_explicit_instantiation' attribute ignored on a non-template context}}
   };
   template <class T>
-  struct EXCLUDE_ATTR class_template_excluded {};
-  // expected-warning@-1{{'exclude_from_explicit_instantiation' attribute ignored on a non-template context}}
+  struct EXCLUDE_ATTR class_template_excluded {}; // expected-warning{{'exclude_from_explicit_instantiation' attribute ignored on a non-template context}}
   template <class T>
-  EXCLUDE_ATTR static T var_template_excluded;
-  // expected-warning@-1{{'exclude_from_explicit_instantiation' attribute ignored on a non-template context}}
+  EXCLUDE_ATTR static T var_template_excluded; // expected-warning{{'exclude_from_explicit_instantiation' attribute ignored on a non-template context}}
   template <class T>
-  EXCLUDE_ATTR void fn_template_excluded();
-  // expected-warning@-1{{'exclude_from_explicit_instantiation' attribute ignored on a non-template context}}
+  EXCLUDE_ATTR void fn_template_excluded(); // expected-warning{{'exclude_from_explicit_instantiation' attribute ignored on a non-template context}}
 };
 
-struct EXCLUDE_ATTR class_excluded {};
-// expected-warning@-1{{'exclude_from_explicit_instantiation' attribute ignored on a non-member declaration}}
-EXCLUDE_ATTR int var_excluded;
-// expected-warning@-1{{'exclude_from_explicit_instantiation' attribute ignored on a non-member declaration}}
-EXCLUDE_ATTR void fn_excluded();
-// expected-warning@-1{{'exclude_from_explicit_instantiation' attribute ignored on a non-member declaration}}
+struct EXCLUDE_ATTR class_excluded {}; // expected-warning{{'exclude_from_explicit_instantiation' attribute ignored on a non-member declaration}}
+EXCLUDE_ATTR int var_excluded; // expected-warning{{'exclude_from_explicit_instantiation' attribute ignored on a non-member declaration}}
+EXCLUDE_ATTR void fn_excluded(); // expected-warning{{'exclude_from_explicit_instantiation' attribute ignored on a non-member declaration}}
 
 template <class T>
-struct EXCLUDE_ATTR class_template_excluded {};
-// expected-warning@-1{{'exclude_from_explicit_instantiation' attribute ignored on a non-member declaration}}
+struct EXCLUDE_ATTR class_template_excluded {}; // expected-warning{{'exclude_from_explicit_instantiation' attribute ignored on a non-member declaration}}
 template <class T>
-EXCLUDE_ATTR T var_template_excluded;
-// expected-warning@-1{{'exclude_from_explicit_instantiation' attribute ignored on a non-member declaration}}
+EXCLUDE_ATTR T var_template_excluded; // expected-warning{{'exclude_from_explicit_instantiation' attribute ignored on a non-member declaration}}
 template <class T>
-EXCLUDE_ATTR void fn_template_excluded();
-// expected-warning@-1{{'exclude_from_explicit_instantiation' attribute ignored on a non-member declaration}}
+EXCLUDE_ATTR void fn_template_excluded(); // expected-warning{{'exclude_from_explicit_instantiation' attribute ignored on a non-member declaration}}
 
 void fn () {
-  EXCLUDE_ATTR static int var_excluded;
-  // expected-warning@-1{{'exclude_from_explicit_instantiation' attribute ignored on a non-member declaration}}
+  EXCLUDE_ATTR static int var_excluded; // expected-warning{{'exclude_from_explicit_instantiation' attribute ignored on a non-member declaration}}
 }
 
 auto lambda = [](auto x) {
-  EXCLUDE_ATTR static int var_excluded;
-  // expected-warning@-1{{'exclude_from_explicit_instantiation' attribute ignored on a non-member declaration}}
+  EXCLUDE_ATTR static int var_excluded; // expected-warning{{'exclude_from_explicit_instantiation' attribute ignored on a non-member declaration}}
 };
 
 template <class T>
 void fn_template() {
-  EXCLUDE_ATTR static T var_excluded;
-  // expected-warning@-1{{'exclude_from_explicit_instantiation' attribute ignored on a non-member declaration}}
+  EXCLUDE_ATTR static T var_excluded; // expected-warning{{'exclude_from_explicit_instantiation' attribute ignored on a non-member declaration}}
 };
-- 
2.53.0

