From 5de04d6fd9e2921c2cad078fe830a129aafd9bad Mon Sep 17 00:00:00 2001
From: kikairoya <kikairoya@gmail.com>
Date: Tue, 7 Apr 2026 18:25:24 +0900
Subject: [PATCH 10/54] registry

---
 llvm/include/llvm/Support/Registry.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/llvm/include/llvm/Support/Registry.h b/llvm/include/llvm/Support/Registry.h
index e9ad4c9819ca2bf715abb290f31e125c22dcf96e..35a5a3e324e267aea6469826e620f6dc291bc1bc 100644
--- a/llvm/include/llvm/Support/Registry.h
+++ b/llvm/include/llvm/Support/Registry.h
@@ -217,7 +217,7 @@ public:
 #if defined(_MSC_VER) && !defined(__clang__) &&                                \
     defined(LLVM_ENABLE_LLVM_EXPORT_ANNOTATIONS) &&                            \
     !defined(LLVM_BUILD_STATIC)
-#define LLVM_DETAIL_DECLARE_REGISTRY_IMPORT __declspec(dllimport)
+#define LLVM_DETAIL_DECLARE_REGISTRY_IMPORT __declspec(dllexport)
 #define LLVM_DETAIL_DEFINE_REGISTRY_SUPPRES_WARNING_BEGIN                      \
   _Pragma("warning(push)") _Pragma("warning(disable:4273)")
 #define LLVM_DETAIL_DEFINE_REGISTRY_SUPPRES_WARNING_END _Pragma("warning(pop)")
@@ -235,7 +235,7 @@ public:
   struct RegistryLinkListDeclarationMarker<REGISTRY_CLASS> : std::true_type {  \
   };                                                                           \
   template <>                                                                  \
-  LLVM_DETAIL_DECLARE_REGISTRY_IMPORT                                          \
+  /*LLVM_DETAIL_DECLARE_REGISTRY_IMPORT*/LLVM_ABI_EXPORT                               \
       RegistryLinkListStorage<REGISTRY_CLASS> &                                \
       getRegistryLinkListInstance<REGISTRY_CLASS>();                           \
   }
-- 
2.53.0

