From fb150ee07d05993d0698b226098cd7f5b5ac175c Mon Sep 17 00:00:00 2001
From: kikairoya <kikairoya@gmail.com>
Date: Mon, 3 Nov 2025 17:11:21 +0900
Subject: [PATCH 23/39] drop -Wl,--export-all-symbols automatically added by
 CMake

---
 llvm/CMakeLists.txt | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/llvm/CMakeLists.txt b/llvm/CMakeLists.txt
index 908580f791f361addcd23eb4e4af03a09bf91765..f63a6909e77550b1ebc7dc445b5e15017649a38a 100644
--- a/llvm/CMakeLists.txt
+++ b/llvm/CMakeLists.txt
@@ -1308,6 +1308,15 @@ endif( "${CMAKE_SYSTEM_NAME}" MATCHES SunOS )
 # use EXPORT_SYMBOLS argument.
 set(CMAKE_SHARED_LIBRARY_LINK_CXX_FLAGS "")
 
+if (CYGWIN AND LLVM_EXPORT_SYMBOLS_FOR_PLUGINS)
+  # CMake sets CMAKE_EXE_EXPORTS_${lang}_FLAG to `-Wl,--export-all-symbols`
+  # for Cygwin. This is incompatible with LLVM_EXPORT_SYMBOLS_FOR_PLUGINS
+  # tweaks, so unset them.
+  set(CMAKE_EXE_EXPORTS_ASM_FLAG "")
+  set(CMAKE_EXE_EXPORTS_C_FLAG "")
+  set(CMAKE_EXE_EXPORTS_CXX_FLAG "")
+endif()
+
 include(AddLLVM)
 include(TableGen)
 
-- 
2.51.2

