From d66f0baa9d7bd29e03e10cc72b5693a54543f4c5 Mon Sep 17 00:00:00 2001
From: kikairoya <kikairoya@gmail.com>
Date: Sun, 7 Sep 2025 22:39:48 +0900
Subject: [PATCH 28/52] update test

---
 .../CodeGenCXX/mingw-template-dllexport.cpp   | 52 +++++++------------
 1 file changed, 20 insertions(+), 32 deletions(-)

diff --git a/clang/test/CodeGenCXX/mingw-template-dllexport.cpp b/clang/test/CodeGenCXX/mingw-template-dllexport.cpp
index ac9f060c9569aa84cb8adb54a5f48c1ccc0d2650..5bf70b625d1467ffee78d980e06213bc17249f95 100644
--- a/clang/test/CodeGenCXX/mingw-template-dllexport.cpp
+++ b/clang/test/CodeGenCXX/mingw-template-dllexport.cpp
@@ -167,30 +167,18 @@ template <class T> int outer<T>::inner::v = 0;
 // FUNC:     define{{.*}} dllexport{{.*}} @_ZN5outerIiE1fEv
 // FUNC:     define{{.*}} dllexport{{.*}} @_ZN5outerIiE1gEv
 // FUNC-NOT: define{{.*}} @_ZN5outerIiE1hEv
-// FUNC-NOT: define{{.*}} dllexport{{.*}} @_ZN5outerIiE5innerC2Ev
-// FUNC:     define{{.*}} dso_local{{.*}} @_ZN5outerIiE5innerC2Ev
-// FUNC-NOT: define{{.*}} dllexport{{.*}} @_ZN5outerIiE5innerC1Ev
-// FUNC:     define{{.*}} dso_local{{.*}} @_ZN5outerIiE5innerC1Ev
-// FUNC-NOT: define{{.*}} dllexport{{.*}} @_ZN5outerIiE5innerC2ERKS1_
-// FUNC:     define{{.*}} dso_local{{.*}} @_ZN5outerIiE5innerC2ERKS1_
-// FUNC-NOT: define{{.*}} dllexport{{.*}} @_ZN5outerIiE5innerC1ERKS1_
-// FUNC:     define{{.*}} dso_local{{.*}} @_ZN5outerIiE5innerC1ERKS1_
-// FUNC-NOT: define{{.*}} dllexport{{.*}} @_ZN5outerIiE5innerC2EOS1_
-// FUNC:     define{{.*}} dso_local{{.*}} @_ZN5outerIiE5innerC2EOS1_
-// FUNC-NOT: define{{.*}} dllexport{{.*}} @_ZN5outerIiE5innerC1EOS1_
-// FUNC:     define{{.*}} dso_local{{.*}} @_ZN5outerIiE5innerC1EOS1_
-// FUNC-NOT: define{{.*}} dllexport{{.*}} @_ZN5outerIiE5innerD2Ev
-// FUNC:     define{{.*}} dso_local{{.*}} @_ZN5outerIiE5innerD2Ev
-// FUNC-NOT: define{{.*}} dllexport{{.*}} @_ZN5outerIiE5innerD1Ev
-// FUNC:     define{{.*}} dso_local{{.*}} @_ZN5outerIiE5innerD1Ev
-// FUNC-NOT: define{{.*}} dllexport{{.*}} @_ZN5outerIiE5inneraSERKS1_
-// FUNC:     define{{.*}} dso_local{{.*}} @_ZN5outerIiE5inneraSERKS1_
-// FUNC-NOT: define{{.*}} dllexport{{.*}} @_ZN5outerIiE5inneraSEOS1_
-// FUNC:     define{{.*}} dso_local{{.*}} @_ZN5outerIiE5inneraSEOS1_
-// FUNC-NOT: define{{.*}} dllexport{{.*}} @_ZN5outerIiE5inner1fEv
-// FUNC:     define{{.*}} dso_local{{.*}} @_ZN5outerIiE5inner1fEv
-// FUNC-NOT: define{{.*}} dllexport{{.*}} @_ZN5outerIiE5inner1gEv
-// FUNC:     define{{.*}} dso_local{{.*}} @_ZN5outerIiE5inner1gEv
+// FUNC:     define{{.*}} dllexport{{.*}} @_ZN5outerIiE5innerC2Ev
+// FUNC:     define{{.*}} dllexport{{.*}} @_ZN5outerIiE5innerC1Ev
+// FUNC:     define{{.*}} dllexport{{.*}} @_ZN5outerIiE5innerC2ERKS1_
+// FUNC:     define{{.*}} dllexport{{.*}} @_ZN5outerIiE5innerC1ERKS1_
+// FUNC:     define{{.*}} dllexport{{.*}} @_ZN5outerIiE5innerC2EOS1_
+// FUNC:     define{{.*}} dllexport{{.*}} @_ZN5outerIiE5innerC1EOS1_
+// FUNC:     define{{.*}} dllexport{{.*}} @_ZN5outerIiE5innerD2Ev
+// FUNC:     define{{.*}} dllexport{{.*}} @_ZN5outerIiE5innerD1Ev
+// FUNC:     define{{.*}} dllexport{{.*}} @_ZN5outerIiE5inneraSERKS1_
+// FUNC:     define{{.*}} dllexport{{.*}} @_ZN5outerIiE5inneraSEOS1_
+// FUNC:     define{{.*}} dllexport{{.*}} @_ZN5outerIiE5inner1fEv
+// FUNC:     define{{.*}} dllexport{{.*}} @_ZN5outerIiE5inner1gEv
 // FUNC-NOT: define{{.*}} @_ZN5outerIiE5inner1hEv
 template struct __declspec(dllexport) outer<int>;
 
@@ -206,18 +194,18 @@ template struct __declspec(dllexport) outer<int>;
 // FUNC-O0:  declare{{.*}} dllimport{{.*}} @_ZN5outerIcEC1EOS0_
 // FUNC-O0:  declare{{.*}} dllimport{{.*}} @_ZN5outerIcEaSERKS0_
 // FUNC-O0:  declare{{.*}} dllimport{{.*}} @_ZN5outerIcEaSEOS0_
-// FUNC-O0:  declare{{.*}} dso_local{{.*}} @_ZN5outerIcE5innerC1Ev
-// FUNC-O0:  declare{{.*}} dso_local{{.*}} @_ZN5outerIcE5innerC1ERKS1_
-// FUNC-O0:  declare{{.*}} dso_local{{.*}} @_ZN5outerIcE5innerC1EOS1_
-// FUNC-O0:  declare{{.*}} dso_local{{.*}} @_ZN5outerIcE5inneraSERKS1_
-// FUNC-O0:  declare{{.*}} dso_local{{.*}} @_ZN5outerIcE5inneraSEOS1_
+// FUNC-O0:  declare{{.*}} dllimport{{.*}} @_ZN5outerIcE5innerC1Ev
+// FUNC-O0:  declare{{.*}} dllimport{{.*}} @_ZN5outerIcE5innerC1ERKS1_
+// FUNC-O0:  declare{{.*}} dllimport{{.*}} @_ZN5outerIcE5innerC1EOS1_
+// FUNC-O0:  declare{{.*}} dllimport{{.*}} @_ZN5outerIcE5inneraSERKS1_
+// FUNC-O0:  declare{{.*}} dllimport{{.*}} @_ZN5outerIcE5inneraSEOS1_
 // FUNC-O0:  declare{{.*}} dllimport{{.*}} @_ZN5outerIcE1fEv
 // FUNC:     declare{{.*}} dllimport{{.*}} @_ZN5outerIcE1gEv
 // FUNC-O0:  define{{.*}} dso_local{{.*}} @_ZN5outerIcE1hEv
-// FUNC-O0:  declare{{.*}} dso_local{{.*}} @_ZN5outerIcE5inner1fEv
-// FUNC:     declare{{.*}} dso_local{{.*}} @_ZN5outerIcE5inner1gEv
+// FUNC-O0:  declare{{.*}} dllimport{{.*}} @_ZN5outerIcE5inner1fEv
+// FUNC:     declare{{.*}} dllimport{{.*}} @_ZN5outerIcE5inner1gEv
 // FUNC-O0:  define{{.*}} dso_local{{.*}} @_ZN5outerIcE5inner1hEv
-// FUNC-O0:  declare{{.*}} dso_local{{.*}} @_ZN5outerIcE5innerD1Ev
+// FUNC-O0:  declare{{.*}} dllimport{{.*}} @_ZN5outerIcE5innerD1Ev
 // FUNC-O0:  declare{{.*}} dllimport{{.*}} @_ZN5outerIcED1Ev
 extern template struct __declspec(dllimport) outer<char>;
 
-- 
2.51.2

