From a1e03d81454fcd14cbca9823a7befc1609e790cd Mon Sep 17 00:00:00 2001
From: kikairoya <kikairoya@gmail.com>
Date: Sun, 2 Nov 2025 16:32:02 +0900
Subject: [PATCH 32/46] pretest

---
 .../dllexport-no-dllexport-inlines.cpp         | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/clang/test/CodeGenCXX/dllexport-no-dllexport-inlines.cpp b/clang/test/CodeGenCXX/dllexport-no-dllexport-inlines.cpp
index a74d84b3d2d0b9d041a207d898a46177c6943b95..15ec6798563acc642d542075fea64db44a850fae 100644
--- a/clang/test/CodeGenCXX/dllexport-no-dllexport-inlines.cpp
+++ b/clang/test/CodeGenCXX/dllexport-no-dllexport-inlines.cpp
@@ -49,6 +49,15 @@ struct __declspec(dllexport) ExportedClass {
   // MINGW-DAG: define linkonce_odr dso_local noundef i32 @_ZN13ExportedClass39InlineOutclassDefFuncWithStaticVariableEv
   inline int InlineOutclassDefFuncWithStaticVariable();
 
+  // NOEXPORTINLINE-DAG: define weak_odr dso_local dllexport void @"?InlineLaterOutclassDefFunc@ExportedClass@@QEAAXXZ
+  // EXPORTINLINE-DAG: define weak_odr dso_local dllexport void @"?InlineLaterOutclassDefFunc@ExportedClass@@QEAAXXZ
+  // MINGW-DAG: define weak_odr dso_local dllexport void @_ZN13ExportedClass26InlineLaterOutclassDefFuncEv
+  void InlineLaterOutclassDefFunc();
+
+  // CHECK-DAG: define weak_odr dso_local dllexport noundef i32 @"?InlineLaterOutclassDefFuncWithStaticVariable@ExportedClass@@QEAAHXZ"
+  // MINGW-DAG: define weak_odr dso_local dllexport noundef i32 @_ZN13ExportedClass44InlineLaterOutclassDefFuncWithStaticVariableEv
+  int InlineLaterOutclassDefFuncWithStaticVariable();
+
   // CHECK-DAG: define dso_local dllexport void @"?OutoflineDefFunc@ExportedClass@@QEAAXXZ"
   // MINGW-DAG: define dso_local dllexport void @_ZN13ExportedClass16OutoflineDefFuncEv
   void OutoflineDefFunc();
@@ -64,14 +73,23 @@ inline int ExportedClass::InlineOutclassDefFuncWithStaticVariable() {
   return ++static_variable;
 }
 
+inline void ExportedClass::InlineLaterOutclassDefFunc() {}
+
+inline int ExportedClass::InlineLaterOutclassDefFuncWithStaticVariable() {
+  static int static_variable = 0;
+  return ++static_variable;
+}
+
 void ExportedClassUser() {
   ExportedClass a;
   a.InclassDefFunc();
   a.InlineOutclassDefFunc();
+  a.InlineLaterOutclassDefFunc();
 #if defined(__MINGW32__)
   a.InclassDefFuncWithStaticVariable();
   a.InclassDefFuncWithLambdaStaticVariable();
   a.InlineOutclassDefFuncWithStaticVariable();
+  a.InlineLaterOutclassDefFuncWithStaticVariable();
 #endif
 }
 
-- 
2.51.2

