From fabb7b68921b223d6ace7ab14ca68cd2d3d0bf23 Mon Sep 17 00:00:00 2001
From: kikairoya <kikairoya@gmail.com>
Date: Fri, 20 Mar 2026 18:24:43 +0900
Subject: [PATCH 09/62] take account of MSVC dllexport rule

---
 llvm/include/llvm/Support/Registry.h | 26 ++++++++++++++++++++++++--
 1 file changed, 24 insertions(+), 2 deletions(-)

diff --git a/llvm/include/llvm/Support/Registry.h b/llvm/include/llvm/Support/Registry.h
index e9b3c24977e3cdecbbbda14b58e7f96f063f4b07..e9ad4c9819ca2bf715abb290f31e125c22dcf96e 100644
--- a/llvm/include/llvm/Support/Registry.h
+++ b/llvm/include/llvm/Support/Registry.h
@@ -208,6 +208,25 @@ public:
 
 } // end namespace llvm
 
+// MSVC disallows to add dllexport/dllimport after first declaration but allows
+// to change dllimport to dllexport. Therefore add dllimport always to the
+// specialization declaration and then override by dllexport on the definition
+// with ignoring the warning about doing that.
+// In contrast, Clang allows to add dllexport to specialization definitions
+// while unable to override by later dllexport.
+#if defined(_MSC_VER) && !defined(__clang__) &&                                \
+    defined(LLVM_ENABLE_LLVM_EXPORT_ANNOTATIONS) &&                            \
+    !defined(LLVM_BUILD_STATIC)
+#define LLVM_DETAIL_DECLARE_REGISTRY_IMPORT __declspec(dllimport)
+#define LLVM_DETAIL_DEFINE_REGISTRY_SUPPRES_WARNING_BEGIN                      \
+  _Pragma("warning(push)") _Pragma("warning(disable:4273)")
+#define LLVM_DETAIL_DEFINE_REGISTRY_SUPPRES_WARNING_END _Pragma("warning(pop)")
+#else
+#define LLVM_DETAIL_DECLARE_REGISTRY_IMPORT
+#define LLVM_DETAIL_DEFINE_REGISTRY_SUPPRES_WARNING_BEGIN
+#define LLVM_DETAIL_DEFINE_REGISTRY_SUPPRES_WARNING_END
+#endif
+
 /// Helper macro to declare registry class.
 ///
 #define LLVM_DECLARE_REGISTRY(REGISTRY_CLASS)                                  \
@@ -216,8 +235,9 @@ public:
   struct RegistryLinkListDeclarationMarker<REGISTRY_CLASS> : std::true_type {  \
   };                                                                           \
   template <>                                                                  \
-  RegistryLinkListStorage<REGISTRY_CLASS> &                                    \
-  getRegistryLinkListInstance<REGISTRY_CLASS>();                               \
+  LLVM_DETAIL_DECLARE_REGISTRY_IMPORT                                          \
+      RegistryLinkListStorage<REGISTRY_CLASS> &                                \
+      getRegistryLinkListInstance<REGISTRY_CLASS>();                           \
   }
 
 /// Helper macro to define registry class.
@@ -233,12 +253,14 @@ public:
                 "Missing matching registry delcaration of " #REGISTRY_CLASS    \
                 ". Place `LLVM_DECLARE_REGISTRY(" #REGISTRY_CLASS              \
                 ")` in a header.");                                            \
+  LLVM_DETAIL_DEFINE_REGISTRY_SUPPRES_WARNING_BEGIN                            \
   template <>                                                                  \
   LLVM_ABI_EXPORT RegistryLinkListStorage<REGISTRY_CLASS> &                    \
   getRegistryLinkListInstance<REGISTRY_CLASS>() {                              \
     static RegistryLinkListStorage<REGISTRY_CLASS> Instance;                   \
     return Instance;                                                           \
   }                                                                            \
+  LLVM_DETAIL_DEFINE_REGISTRY_SUPPRES_WARNING_END                              \
   }
 
 #define LLVM_DETAIL_INSTANTIATE_REGISTRY_1(ABITAG, REGISTRY_CLASS)             \
-- 
2.53.0

