From 7c92206814a072d9c0e91245fe56290a90906baf Mon Sep 17 00:00:00 2001
From: kikairoya <kikairoya@gmail.com>
Date: Mon, 3 Nov 2025 17:11:21 +0900
Subject: [PATCH 30/56] drop -Wl,--export-all-symbols automatically added by
 CMake

---
 llvm/CMakeLists.txt | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/llvm/CMakeLists.txt b/llvm/CMakeLists.txt
index 6d94cbbcd2559d3b191bbe336aae6b7e92e00a0c..e08f5dd5ca4b2e46c535d71418a864d99db3182e 100644
--- a/llvm/CMakeLists.txt
+++ b/llvm/CMakeLists.txt
@@ -1309,6 +1309,15 @@ endif( "${CMAKE_SYSTEM_NAME}" MATCHES SunOS )
 # use EXPORT_SYMBOLS argument.
 set(CMAKE_SHARED_LIBRARY_LINK_CXX_FLAGS "")
 
+if (CYGWIN AND LLVM_EXPORT_SYMBOLS_FOR_PLUGINS)
+  # CMake sets CMAKE_EXE_EXPORTS_${lang}_FLAG to `-Wl,--export-all-symbols`
+  # for Cygwin. This is incompatible with LLVM_EXPORT_SYMBOLS_FOR_PLUGINS
+  # tweaks, so unset them.
+  set(CMAKE_EXE_EXPORTS_ASM_FLAG "")
+  set(CMAKE_EXE_EXPORTS_C_FLAG "")
+  set(CMAKE_EXE_EXPORTS_CXX_FLAG "")
+endif()
+
 include(AddLLVM)
 include(TableGen)
 
-- 
2.52.0

