From 49d49d9308d0ec803a7de4caaf918cdbd17c5b93 Mon Sep 17 00:00:00 2001
From: kikairoya <kikairoya@gmail.com>
Date: Wed, 26 Aug 2026 20:42:26 +0900
Subject: [PATCH 35/64] [LLVM][SPIRV] Add a null check

---
 llvm/lib/Target/SPIRV/SPIRVGlobalRegistry.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/llvm/lib/Target/SPIRV/SPIRVGlobalRegistry.cpp b/llvm/lib/Target/SPIRV/SPIRVGlobalRegistry.cpp
index a0eab4a541550f1a82870179797c8909a0c68036..faea6fcd97fbe43a8ccc35b01a054798dfd79727 100644
--- a/llvm/lib/Target/SPIRV/SPIRVGlobalRegistry.cpp
+++ b/llvm/lib/Target/SPIRV/SPIRVGlobalRegistry.cpp
@@ -2362,6 +2362,7 @@ void SPIRVGlobalRegistry::buildAssignPtr(IRBuilder<> &B, Type *ElemTy,
   Function *CurrF =
       B.GetInsertBlock() ? B.GetInsertBlock()->getParent() : nullptr;
   if (AssignPtrTyCI == nullptr ||
+      AssignPtrTyCI->getParent() == nullptr ||
       AssignPtrTyCI->getParent()->getParent() != CurrF) {
     AssignPtrTyCI = buildIntrWithMD(
         Intrinsic::spv_assign_ptr_type, {Arg->getType()}, OfType, Arg,
-- 
2.55.0

