From 4a46e3d4abc46ff7f93fcd7e0bc98faf885cd147 Mon Sep 17 00:00:00 2001
From: kikairoya <kikairoya@gmail.com>
Date: Sat, 14 Jun 2025 08:15:58 +0900
Subject: [PATCH 43/64] [LLVM-Tools][Cygwin] Enable LTO component

---
 llvm/tools/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/llvm/tools/CMakeLists.txt b/llvm/tools/CMakeLists.txt
index 0710d54fad22a999347d79ef5047c68650798c1e..2276a8af1f7c314a9493979439528a5d86d84344 100644
--- a/llvm/tools/CMakeLists.txt
+++ b/llvm/tools/CMakeLists.txt
@@ -17,7 +17,7 @@ if(NOT LLVM_USE_INTEL_JITEVENTS )
   set(LLVM_TOOL_LLVM_JITLISTENER_BUILD Off)
 endif()
 
-if(CYGWIN OR NOT LLVM_ENABLE_PIC)
+if(NOT LLVM_ENABLE_PIC)
   set(LLVM_TOOL_LTO_BUILD Off)
 endif()
 
-- 
2.54.0

