From 08a90989f66a11f7788bc738954c644bbeb7aa7b Mon Sep 17 00:00:00 2001
From: kikairoya <kikairoya@gmail.com>
Date: Sun, 4 Jan 2026 21:53:48 +0900
Subject: [PATCH 44/65] [Clang][Cygwin] Set default DWARF version to 4

---
 clang/lib/Driver/ToolChains/Cygwin.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/clang/lib/Driver/ToolChains/Cygwin.h b/clang/lib/Driver/ToolChains/Cygwin.h
index 4371fee81d07c41190f0422e2f695446249fdb2f..934110c1dd4c3649a8a20c6849853139991f092a 100644
--- a/clang/lib/Driver/ToolChains/Cygwin.h
+++ b/clang/lib/Driver/ToolChains/Cygwin.h
@@ -33,6 +33,8 @@ public:
                         llvm::opt::ArgStringList &CC1Args, BoundArch BA,
                         Action::OffloadKind DeviceOffloadKind) const override;
 
+  unsigned GetDefaultDwarfVersion() const override { return 4; }
+
 protected:
   Tool *buildLinker() const override;
 };
-- 
2.55.0

