From 5e993342af93e010dfdbaf75194244bd7de480a7 Mon Sep 17 00:00:00 2001
From: kikairoya <kikairoya@gmail.com>
Date: Sun, 4 Jan 2026 21:53:48 +0900
Subject: [PATCH 44/63] [Clang][Cygwin] Set default DWARF version to 4

---
 clang/lib/Driver/ToolChains/Cygwin.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/clang/lib/Driver/ToolChains/Cygwin.h b/clang/lib/Driver/ToolChains/Cygwin.h
index a597de8f992711e42617e4da9073d1a59e7fb7e5..b81793e3d5e29b126e0ba300c8982146927dedb3 100644
--- a/clang/lib/Driver/ToolChains/Cygwin.h
+++ b/clang/lib/Driver/ToolChains/Cygwin.h
@@ -34,6 +34,8 @@ public:
                         llvm::StringRef BoundArch,
                         Action::OffloadKind DeviceOffloadKind) const override;
 
+  unsigned GetDefaultDwarfVersion() const override { return 4; }
+
 protected:
   Tool *buildLinker() const override;
 };
-- 
2.54.0

