From 2209779d0e7ec749cb805245548bfc078ffe0017 Mon Sep 17 00:00:00 2001
From: kikairoya <kikairoya@gmail.com>
Date: Tue, 3 Feb 2026 19:54:03 +0900
Subject: [PATCH 47/70] don't use Cygwin provided stdatomic.h

---
 clang/lib/Headers/stdatomic.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/clang/lib/Headers/stdatomic.h b/clang/lib/Headers/stdatomic.h
index 1991351f9e9efed0fc4154afe02c9a9faf200c73..76e5da0eef36562c7612e8e6819b42e1e98bf78a 100644
--- a/clang/lib/Headers/stdatomic.h
+++ b/clang/lib/Headers/stdatomic.h
@@ -19,6 +19,7 @@
  * `stdatomic.h` header requires C++23 or newer.
  */
 #if __STDC_HOSTED__ &&                                                         \
+    !defined(__CYGWIN__) &&                                                    \
     __has_include_next(<stdatomic.h>) &&                                       \
     (!defined(_MSC_VER) || (defined(__cplusplus) && __cplusplus >= 202002L))
 # include_next <stdatomic.h>
-- 
2.52.0

