From 530dfd2d855830841faf2677b404ecce3617f437 Mon Sep 17 00:00:00 2001
From: kikairoya <kikairoya@gmail.com>
Date: Sat, 18 Jul 2026 15:29:18 +0900
Subject: [PATCH 24/38] [LLVM] Disable plugin support by default on Windows

---
 llvm/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/llvm/CMakeLists.txt b/llvm/CMakeLists.txt
index 28e96fc28b5aa660fe821507cf78a490abaad973..8cd2aed147f38029c9df4af89e441e9f12d5ff5f 100644
--- a/llvm/CMakeLists.txt
+++ b/llvm/CMakeLists.txt
@@ -1113,7 +1113,7 @@ message(STATUS "LLVM default target triple: ${LLVM_DEFAULT_TARGET_TRIPLE}")
 set(LLVM_TARGET_TRIPLE "${LLVM_DEFAULT_TARGET_TRIPLE}")
 
 if(WIN32 OR CYGWIN)
-  if(BUILD_SHARED_LIBS OR LLVM_BUILD_LLVM_DYLIB)
+  if(BUILD_SHARED_LIBS OR LLVM_LINK_LLVM_DYLIB)
     set(LLVM_ENABLE_PLUGINS_default ON)
   else()
     set(LLVM_ENABLE_PLUGINS_default OFF)
-- 
2.55.0

