From 770886245b8cc884ea6ad2d7c1d8cb6dac081650 Mon Sep 17 00:00:00 2001
From: kikairoya <kikairoya@gmail.com>
Date: Fri, 31 Jul 2026 22:57:24 +0900
Subject: [PATCH 29/38] Link dependants with PLUGIN_TOOL more greedy

---
 llvm/cmake/modules/AddLLVM.cmake | 66 +++++++++++++++-----------------
 1 file changed, 31 insertions(+), 35 deletions(-)

diff --git a/llvm/cmake/modules/AddLLVM.cmake b/llvm/cmake/modules/AddLLVM.cmake
index 88b34b1289bc3e2ea5095f36c4f6924f59b240cb..e4aa3452522a76d1b7138c511d3e7d296300b882 100644
--- a/llvm/cmake/modules/AddLLVM.cmake
+++ b/llvm/cmake/modules/AddLLVM.cmake
@@ -1501,7 +1501,11 @@ function(do_export_executable_symbols_for_plugins target exported_symbol_file)
         else()
           list(APPEND other_libs ${lib})
         endif()
-        get_target_property(transitive_libs ${lib} INTERFACE_LINK_LIBRARIES)
+        if(WIN32 OR CYGWIN)
+          get_target_property(transitive_libs ${lib} LINK_LIBRARIES)
+        else()
+          get_target_property(transitive_libs ${lib} INTERFACE_LINK_LIBRARIES)
+        endif()
         foreach(transitive_lib ${transitive_libs})
           if(TARGET ${transitive_lib} AND NOT ${transitive_lib} IN_LIST link_libs)
             list(APPEND newer_libs ${transitive_lib})
@@ -1514,31 +1518,33 @@ function(do_export_executable_symbols_for_plugins target exported_symbol_file)
     set(newer_libs "")
   endwhile()
   list(REMOVE_DUPLICATES static_libs)
-  if(static_libs)
-    if (MSVC OR CMAKE_CXX_SIMULATE_ID STREQUAL "MSVC")
-      set(mangling microsoft)
+  if(exported_symbol_file)
+    if(static_libs)
+      if (MSVC OR CMAKE_CXX_SIMULATE_ID STREQUAL "MSVC")
+        set(mangling microsoft)
+      else()
+        set(mangling itanium)
+      endif()
+      get_host_tool_path(llvm-nm LLVM_NM llvm_nm_exe llvm_nm_target)
+      get_host_tool_path(llvm-readobj LLVM_READOBJ llvm_readobj_exe llvm_readobj_target)
+      add_custom_command(OUTPUT ${exported_symbol_file}
+                         COMMAND "${Python3_EXECUTABLE}"
+                           ${LLVM_MAIN_SRC_DIR}/utils/extract_symbols.py
+                           --mangling=${mangling} ${static_libs}
+                           -o ${exported_symbol_file}
+                           --nm=${llvm_nm_exe}
+                           --readobj=${llvm_readobj_exe}
+                         WORKING_DIRECTORY ${LLVM_LIBRARY_OUTPUT_INTDIR}
+                         DEPENDS ${LLVM_MAIN_SRC_DIR}/utils/extract_symbols.py
+                           ${static_libs} ${llvm_nm_target} ${llvm_readobj_target}
+                         VERBATIM
+                         COMMENT "Generating export list for ${target}")
     else()
-      set(mangling itanium)
-    endif()
-    get_host_tool_path(llvm-nm LLVM_NM llvm_nm_exe llvm_nm_target)
-    get_host_tool_path(llvm-readobj LLVM_READOBJ llvm_readobj_exe llvm_readobj_target)
-    add_custom_command(OUTPUT ${exported_symbol_file}
-                       COMMAND "${Python3_EXECUTABLE}"
-                         ${LLVM_MAIN_SRC_DIR}/utils/extract_symbols.py
-                         --mangling=${mangling} ${static_libs}
-                         -o ${exported_symbol_file}
-                         --nm=${llvm_nm_exe}
-                         --readobj=${llvm_readobj_exe}
-                       WORKING_DIRECTORY ${LLVM_LIBRARY_OUTPUT_INTDIR}
-                       DEPENDS ${LLVM_MAIN_SRC_DIR}/utils/extract_symbols.py
-                         ${static_libs} ${llvm_nm_target} ${llvm_readobj_target}
-                       VERBATIM
-                       COMMENT "Generating export list for ${target}")
-  else()
-    # When shared or dylib builds, static_libs can be empty.
-    file(GENERATE OUTPUT ${exported_symbol_file} CONTENT "")
+      # When shared or dylib builds, static_libs can be empty.
+      file(GENERATE OUTPUT ${exported_symbol_file} CONTENT "")
+    endif()
+    add_llvm_symbol_exports( ${target} ${exported_symbol_file} )
   endif()
-  add_llvm_symbol_exports( ${target} ${exported_symbol_file} )
   # If something links against this executable then we want a
   # transitive link against only the libraries whose symbols
   # we aren't exporting.
@@ -1547,17 +1553,7 @@ endfunction()
 
 function(do_export_all_symbols_for_plugins target)
   target_link_options(${target} PRIVATE "LINKER:--export-all-symbols")
-  get_target_property(libs ${target} LINK_LIBRARIES)
-  set(shared_libs)
-  foreach(lib ${libs})
-    if(TARGET ${lib})
-      get_target_property(lib_type ${lib} TYPE)
-      if(${lib_type} STREQUAL "SHARED_LIBRARY")
-        list(APPEND shared_libs ${lib})
-      endif()
-    endif()
-  endforeach(lib)
-  set_target_properties(${target} PROPERTIES INTERFACE_LINK_LIBRARIES "${shared_libs}")
+  do_export_executable_symbols_for_plugins(${target} "")
 endfunction()
 
 function(export_executable_symbols target)
-- 
2.54.0

