From 680e11ceeffbbad192d3c2203ca0cf48bc2e6e7d Mon Sep 17 00:00:00 2001
From: kikairoya <kikairoya@gmail.com>
Date: Mon, 16 Mar 2026 06:52:39 +0900
Subject: [PATCH 30/38] make llvm/Debuginfod static

---
 llvm/lib/Debuginfod/CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/llvm/lib/Debuginfod/CMakeLists.txt b/llvm/lib/Debuginfod/CMakeLists.txt
index 5b3403b961a4ccf5c50e2f3f39ce882903fe5b43..204f3fbf2b7f4a32bd16912e2a7581970e4a8fc2 100644
--- a/llvm/lib/Debuginfod/CMakeLists.txt
+++ b/llvm/lib/Debuginfod/CMakeLists.txt
@@ -5,7 +5,11 @@ endif()
 
 # Note: This isn't a component, since that could potentially add a libcurl
 # dependency to libLLVM.
-add_llvm_library(LLVMDebuginfod
+set(libtype STATIC)
+if (BUILD_SHARED_LIBS)
+  set(libtype)
+endif()
+add_llvm_library(LLVMDebuginfod ${libtype}
   BuildIDFetcher.cpp
   Debuginfod.cpp
 
-- 
2.53.0

