From 53e6e8658fc6ef38436118e27d8d681fe27196f1 Mon Sep 17 00:00:00 2001
From: kikairoya <kikairoya@gmail.com>
Date: Fri, 19 Sep 2025 00:30:29 +0900
Subject: [PATCH 37/37] [cmake] Install llvm-windres

---
 llvm/cmake/modules/AddLLVM.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/llvm/cmake/modules/AddLLVM.cmake b/llvm/cmake/modules/AddLLVM.cmake
index 7d40d309d538..33fba2450eb5 100644
--- a/llvm/cmake/modules/AddLLVM.cmake
+++ b/llvm/cmake/modules/AddLLVM.cmake
@@ -1455,6 +1455,7 @@ if(NOT LLVM_TOOLCHAIN_TOOLS)
     llvm-profdata
     llvm-profgen
     llvm-symbolizer
+    llvm-windres
     # symlink version of some of above tools that are enabled by
     # LLVM_INSTALL_BINUTILS_SYMLINKS.
     addr2line
-- 
2.51.2

